README.md

```diff
@@ -230,10 +232,14 @@ A hotel booking application in React. Homework for the [CodeYourFuture React mod
 
 **Test:** The 'Submit' button is initially not clickable, and becomes clickable once every field has the correct input.
 
+**Reflection:** Checking for a valid value after each input changes, instead of once at the end of the page, is a much nicer user experience. If you were a real user of this booking system, would you prefer to fill out all of the fields and find out you made a mistake at the end (like in exercise 26), or would you prefer to be find out you made a mistake after each input?
+
 #### 28. Date picker
 
 **Instructions:** Add the [js-datepicker](https://www.npmjs.com/package/js-datepicker) package to your project using `npm install`, and import it at the top of the file. Add different IDs to your 'check in date' and 'check out date' `<input>` elements, then create two date pickers using `const checkInPicker = datepicker(YOUR_ID)` (where `YOUR_ID` is the ID you assigned to your check in/check out date elements). 
 
 **Hint:** Read the [js-datepicker usage guide](https://www.npmjs.com/package/js-datepicker#basic-usage)
 
-**Test:** The date picker appears when you click on the 'check in date' input element.
+**Test:** The date picker appears when you click on the 'check in date' and 'check out date' input elements.
+
+**Reflection:** Using `js-datepicker` in this exercise is a great way to practice installing and working with packages in JavaScript. While HTML has a [native datepicker](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date), using a package will often give you more functions and properties to work with. Packages prevent wasting time 'reinventing the wheel', as someone has already written the code for you! Think about some of the code you have written in this lesson - are there any packages available that might have helped you to complete the exercises? (hint: search 'javascript package' + your idea on Google! e.g. 'javascript package validation')
```

Comment:
I think this is a good start, but I also think this reflection pushes the trainee into thinking this is an awesome choice with no downsides. It's a pragmatic choice and one they will often make, but of course we know that typically components that hijack or recreate native elements are not accessible. (And of course often packages are of poor quality in other ways and cause large problems later on, because nobody ever looked inside them to see how they worked, but that's maybe a consideration for later on.)

The majority of all js datepickers available on npm are not accessible (no idea if this one is as I have not set up this project or tested it and it doesn't mention anything in the readme). I would look for a more measured reflection, with some for and against here.

Reply:
very good points @SallyMcGrath. I've expanded this reflection in 772bf51 to allow trainees to weigh up the pros/cons of using packages, but I think it may be too long now and tangential to the original focus of the lesson. let me know what you think, happy to simplify or remove it